### ISSUE TITLE
Possible Bug: M3 starts laser when idle and when laser mode is on ($32=1)

### ISSUE THREAD

**Issue by aleslukek** [gnea/grbl#593]
I am using grbl 1.1f on genuine Arduino Uno. I am using CNC for milling and laser operation. Spindle is not controlled via grbl, only laser is. 
I am making a pendant that controls gcode sender and sends gcode commands to the computer. 
I was looking into a command that turns on laser at weakest power when laser mode is on $32=1 and idle, just to see it's position more accurately. I am able to do this when laser mode is off ($32=0) and I use "M4 S1" to do this. 

However laser mode ($32=1) only enables laser when moving: G1, G2 or G3 (https://github.com/gnea/grbl/wiki/Grbl-v1.1-Laser-Mode). So M3 or M4 Sxxx doesn't turn it on when idle. So this works as it should.

Ok here it comes. There is a certain scenario that overcomes this automatic safety switch that enables to turn on laser with M3 when cnc is idle.
Commands i used are as follows:

M4 S1
G1 G91 X0.1 F5000
G1 G91 X-0.1 F5000
M5

Now this is when laser is off, and only turns on when G1 command is moving a CNC. Let's continue.

M3

M3 turns it on now, this time also when idle, even though it should not. It works every time for me. But after jogging a bit it doesn't do this anymore (not always). M5 turns it off, and M3 on again. 
I don't think that exactly G1 G91 X0.1 F5000 is at fault, only G1 and then whatever does it. Haven't tested for G2 or G3.

I believe it is a bug, for safety reasons it should not be acting in such way. I am exploiting this bug for laser test fire at weakest power - but would rather have a dedicated gcode command that would fire laser at weakest power.

tldr: when $32=1, M3 turns laser on when idle after G1 command. Would be awesome to have laser test fire.

**Comment by chamnit** [comment]
Allowing the laser to turn on while in M3 is intentionally. Precisely for reason you have been using it for. Test firing, focusing and locating. It’s not a bug. It was fulfilling a request from many longtime laser users.
